### DISCUSSION TITLE
xacro to urdf

### DISCUSSION THREAD

**Discussion by RafaelBrunet** [UniversalRobots/Universal_Robots_ROS2_Driver#459] [Q&A] [answered]
Hi,

I need to convert the xacro file to a urdf. Is there a way to run this command below to convert the file to a urdf?

ros2 run xacro xacro ur.urdf.xacro > ur10e.urdf

Thanks in advance for your help.

**Comment by danzimmerman** [comment, marked as answer]
Yes, but you need to add `ur_type` and, apparently, `name`, like

`ros2 run xacro xacro ur.urdf.xacro ur_type:=ur10e name:=some_robot_name > ur10e.urdf`

You can probably just invoke `xacro` directly like:

`xacro ur.urdf.xacro ur_type:=ur10e name:=some_robot_name > ur10e.urdf`
